cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/batch/ETLBatchTemplate.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.batch;
+
+import co.cask.cdap.api.templates.AdapterConfigurer;
+import co.cask.cdap.api.templates.ApplicationTemplate;
+import co.cask.cdap.internal.schedule.TimeSchedule;
+import co.cask.cdap.templates.etl.api.StageSpecification;
+import co.cask.cdap.templates.etl.api.Transform;
+import co.cask.cdap.templates.etl.api.batch.BatchSink;
+import co.cask.cdap.templates.etl.api.batch.BatchSource;
+import co.cask.cdap.templates.etl.batch.config.ETLBatchConfig;
+import co.cask.cdap.templates.etl.batch.sinks.KVTableSink;
+import co.cask.cdap.templates.etl.batch.sources.KVTableSource;
+import co.cask.cdap.templates.etl.common.Constants;
+import co.cask.cdap.templates.etl.common.DefaultStageConfigurer;
+import co.cask.cdap.templates.etl.common.config.ETLStage;
+import co.cask.cdap.templates.etl.transforms.IdentityTransform;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Table;
+import com.google.gson.Gson;
+
+import java.util.List;
+
+/**
+ * ETL Batch Template.
+ */
+public class ETLBatchTemplate extends ApplicationTemplate<ETLBatchConfig> {
+  private static final Gson GSON = new Gson();
+  private final Table<String, String, String> nameToClass;
+
+  public ETLBatchTemplate() throws Exception {
+    nameToClass = HashBasedTable.create();
+    //TODO: Add classes from Lib here to be available for use in the ETL Adapter. Remove this when
+    //plugins management is completed.
+    initTable(Lists.<Class>newArrayList(KVTableSource.class, KVTableSink.class, IdentityTransform.class));
+  }
+
+  private void initTable(List<Class> classList) throws Exception {
+    for (Class klass : classList) {
+      DefaultStageConfigurer configurer = new DefaultStageConfigurer(klass);
+      if (BatchSource.class.isAssignableFrom(klass)) {
+        BatchSource source = (BatchSource) klass.newInstance();
+        source.configure(configurer);
```

Comment:
shouldn't configure here, as it's called from the constructor.  For example, suppose the source creates a stream in it's configure.